src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.Digest.cs

```diff
@@ -404,7 +405,7 @@ private void Parse(string challenge)
                     string? value = GetNextValue(challenge, parsedIndex, MustValueBeQuoted(key), out parsedIndex);
                     // Ensure value is valid.
                     if (string.IsNullOrEmpty(value)
-                        && (value == null || !key.Equals(Opaque, StringComparison.OrdinalIgnoreCase)))
+                        && (value == null || (!key.Equals(Opaque, StringComparison.OrdinalIgnoreCase) && !key.Equals(Domain, StringComparison.OrdinalIgnoreCase))))
```

Comment:
Seems like the logic here would be clearer if the null and empty check were separated. I.e.:

if (value is null)
  break;

if (string.IsEmpty(value) && ...)
  break;

Reply:
I agree. updated.